persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/message/MessageDAOImpl.kt

```diff
@@ -76,17 +77,20 @@ class MessageDAOImpl(
     @Suppress("ComplexMethod", "LongMethod")
     private fun insertInDB(message: MessageEntity) {
         if (!updateIdIfAlreadyExists(message)) {
-            queries.insertOrIgnoreMessage(
-                id = message.id,
-                conversation_id = message.conversationId,
-                date = message.date,
-                sender_user_id = message.senderUserId,
-                sender_client_id = if (message is MessageEntity.Regular) message.senderClientId else null,
-                visibility = message.visibility,
-                status = message.status,
-                content_type = contentTypeOf(message.content),
-                expects_read_confirmation = if (message is MessageEntity.Regular) message.expectsReadConfirmation else false
-            )
+            if (isAssetMessageUpdate(message)) {
+                updateAssetMessage(message)
+            } else
+                queries.insertOrIgnoreMessage(
+                    id = message.id,
+                    conversation_id = message.conversationId,
+                    date = message.date,
+                    sender_user_id = message.senderUserId,
+                    sender_client_id = if (message is MessageEntity.Regular) message.senderClientId else null,
+                    visibility = message.visibility,
+                    status = message.status,
+                    content_type = contentTypeOf(message.content),
+                    expects_read_confirmation = if (message is MessageEntity.Regular) message.expectsReadConfirmation else false
+                )
```

Comment:
```suggestion
            } else {
                queries.insertOrIgnoreMessage(
                    id = message.id,
                    conversation_id = message.conversationId,
                    date = message.date,
                    sender_user_id = message.senderUserId,
                    sender_client_id = if (message is MessageEntity.Regular) message.senderClientId else null,
                    visibility = message.visibility,
                    status = message.status,
                    content_type = contentTypeOf(message.content),
                    expects_read_confirmation = if (message is MessageEntity.Regular) message.expectsReadConfirmation else false
                )
            }
```

Seems a bit too easy to move this piece of code and breaking the `if/else`.